go/types/struct.go

```diff
@@ -195,13 +209,45 @@ func EscapeStructField(input string) string {
 		buf.WriteString(hs)
 		return buf.String()
 	}
+	output := EscapeFields(input, encode)
+	verifyFieldName(output)
+	return output
 
-	output := ""
-	pattern := headPattern
-	for _, ch := range input {
-		output += encode(string([]rune{ch}), pattern)
-		pattern = tailPattern
+}
+
+// IsValidStructFieldName returns whether the name is valid without as a field name in a struct.
+// Valid names must start with `a-zA-Z` and after that `a-zA-Z0-9_`.
+func IsValidStructFieldName(name string) bool {
+	return fieldNameRe.MatchString(name)
+}
+
+func verifyFieldNames(names []string) {
+	if len(names) == 0 {
+		return
 	}
 
-	return output
+	last := names[0]
+	verifyFieldName(last)
+
+	for i := 1; i < len(names); i++ {
+		verifyFieldName(names[i])
+		if names[i] <= last {
```

Comment:
I think perhaps <= is correct because its a check that  things are ordered least to greatest and not unique so if prev is greater than next  you end up violating the check as well as if it were ==